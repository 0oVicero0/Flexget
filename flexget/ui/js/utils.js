'use strict';

String.prototype.startsWith = function(prefix) {
  return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}

app.menu = [];
app.routes = [];

function register_menu(href, caption, icon, order) {
  href = '/ui/#' + href;
  app.menu.push({href: href, caption: caption, icon: icon, order: order})
}

function register_route(name, url, controller, template) {
  app.routes.push({
    name: name,
    url: url,
    controller: controller,
    template: template
  });
}