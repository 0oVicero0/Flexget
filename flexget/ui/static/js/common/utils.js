'use strict';

String.prototype.startsWith = function(prefix) {
    return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}