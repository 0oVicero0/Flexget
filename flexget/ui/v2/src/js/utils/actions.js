export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
