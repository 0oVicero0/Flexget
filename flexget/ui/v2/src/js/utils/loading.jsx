import Loadable from 'react-loadable';
import Loading from 'components/common/Loader';

export function createAsyncComponent(loader) {
  return Loadable({
    loading: Loading,
    loader,
  });
}

