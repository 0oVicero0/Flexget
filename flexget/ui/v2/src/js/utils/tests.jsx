import React from 'react';
import { MuiThemeProvider } from 'material-ui/styles';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

export function themed(component) {
  return (
    <MuiThemeProvider>
      { component }
    </MuiThemeProvider>
  );
}

export function router(component) {
  return (
    <MemoryRouter>
      { component }
    </MemoryRouter>
  );
}

export function provider(component, state = {}) {
  return (
    <Provider store={mockStore(state)}>
      { component }
    </Provider>
  );
}
