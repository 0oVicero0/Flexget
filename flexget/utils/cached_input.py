import copy
import logging
from flexget.plugin import register_plugin

log = logging.getLogger('cached')


class cached(object):
    """
    Implements transparent caching decorator @cached for inputs.

    Decorator has two parameters

    :name: in which the configuration is present in feeds configuration.
    :key: in which the configuration has the cached resource identifier (ie. url). If the :key: is not
    given or present in the configuration :name: is expected to be a cache name (ie. url)

    Configuration assumptions may make this unusable in some (future) inputs
    """

    cache = {}

    def __init__(self, name, key=None):
        self.name = name
        self.key = key

    def __call__(self, func):

        def wrapped_func(*args, **kwargs):
            # get feed from method parameters
            feed = args[1]
            
            # detect api version
            api_ver = 1
            if len(args) == 3:
                api_ver = 2
            
            if api_ver == 1:
                # get name for a cache from feeds configuration
                if not self.name in feed.config:
                    raise Exception('@cache config name %s is not configured in feed %s' % (self.name, feed.name))
                config = feed.config[self.name]
            else:
                config = args[2]

            log.log(5, 'config: %s' % config)
            log.log(5, 'self.name: %s' % self.name)
            log.log(5, 'self.key: %s' % self.key)
            
            if api_ver == 1:
                if isinstance(config, dict) and self.key in config:
                    name = feed.config[self.name][self.key]
                else:
                    name = feed.config[self.name]
            else:
                if isinstance(config, dict) and self.key in config:
                    name = config[self.key]
                else:
                    name = config

            log.debug('cache name: %s (has: %s)' % (name, ', '.join(self.cache.keys())))

            if name in self.cache:
                # return from the cache
                log.log(5, 'cache hit')
                count = 0
                entries = []
                for entry in self.cache[name]:
                    fresh = copy.deepcopy(entry)
                    entries.append(fresh)
                count += 1
                if count > 0:
                    feed.verbose_progress('Restored %s entries from cache' % count, log)
                return entries
            else:
                log.log(5, 'cache miss')
                # call input event
                response = func(*args, **kwargs)
                # store results to cache
                if api_ver == 1:
                    log.debug('v1 storing to cache %s %s entries' % (name, len(feed.entries)))
                    try:
                        self.cache[name] = copy.deepcopy(feed.entries)
                    except TypeError:
                        # might be caused because of backlog restoring some idiotic stuff, so not neccessarily a bug
                        log.critical('Unable to save feed content into cache, if problem persists longer than a day please report this as a bug')
                else:
                    log.debug('storing to cache %s %s entries' % (name, len(feed.entries)))
                    try:
                        self.cache[name] = copy.deepcopy(response)
                    except TypeError:
                        # might be caused because of backlog restoring some idiotic stuff, so not neccessarily a bug
                        log.critical('Unable to save feed content into cache, if problem persists longer than a day please report this as a bug')
                return response
                    

        return wrapped_func


class CacheClearer(object):

    def on_process_start(self, feed):
        """Internal. Clears the input cache on every process start.
        This is neccessary for webui or otherwise it will only use cache.
        """
        log.debug('clearing cache')
        cached.cache = {}

register_plugin(CacheClearer, 'cache_clearer', builtin=True)
