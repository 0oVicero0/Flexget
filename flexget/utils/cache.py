import hashlib
from pathlib import Path

import requests

from flexget.utils.tools import log


def cached_resource(url, base_dir, force=False, max_size=250, directory='cached_resources'):
    """
    Caches a remote resource to local filesystem. Return a tuple of local file name and mime type, use primarily
    for API/WebUI.

    :param url: Resource URL
    :param force: Does not check for existence of cached resource, fetches the remote URL, ignores directory size limit
    :param max_size: Maximum allowed size of directory, in MB.
    :param directory: Name of directory to use. Default is `cached_resources`
    :return: Tuple of file path and mime type
    """
    mime_type = None
    hashed_name = hashlib.md5(url.encode('utf-8')).hexdigest()
    file_path = Path(base_dir) / directory / hashed_name
    directory = file_path.parent

    if not file_path.exists() or force:
        log.debug('caching %s', url)
        response = requests.get(url)
        response.raise_for_status()
        mime_type = response.headers.get('content-type')
        content = response.content
        directory.mkdir(parents=True, exist_ok=True)

        # Checks directory size and trims if necessary.
        size = dir_size(directory) / (1_024 * 1_024.0)
        if not force:
            while size >= max_size:
                log.debug('directory %s size is over the allowed limit of %s, trimming', size, max_size)
                trim_dir(directory)
                size = dir_size(directory) / (1_024 * 1_024.0)
        file_path.write_bytes(content)

    return str(file_path), mime_type


def dir_size(directory: Path):
    """
    Sums the size of all files in a given dir. Not recursive.

    :param directory: Directory to check
    :return: Summed size of all files in Bytes.
    """
    return sum(file.stat().st_size for file in directory.iterdir())


def trim_dir(directory: Path):
    """
    Removed the least accessed file on a given dir

    :param directory: Directory to check
    """

    def access_time(file: Path):
        return file.stat().st_atime

    files = sorted(directory.iterdir(), key=access_time)
    file_name = directory / files[0]
    log.debug('removing least accessed file: %s', file_name)
    file_name.unlink()
