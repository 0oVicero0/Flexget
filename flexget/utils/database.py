from datetime import datetime
from functools import update_wrapper
from sqlalchemy import extract, func, case
from sqlalchemy.orm import synonym
from flexget.manager import Session
from flexget.utils import qualities


def with_session(func):
    """"Creates a session if one was not passed via keyword argument to the function"""

    def wrapper(*args, **kwargs):
        if not kwargs.get('session'):
            kwargs['session'] = Session(autoflush=True)
            try:
                return func(*args, **kwargs)
            finally:
                kwargs['session'].close()
        else:
            return func(*args, **kwargs)
    return wrapper


def pipe_list_synonym(name):
    """Converts pipe separated text into a list"""

    def getter(self):
        attr = getattr(self, name)
        if attr:
            return attr.strip('|').split('|')

    def setter(self, value):
        if isinstance(value, basestring):
            setattr(self, name, value)
        else:
            setattr(self, name, '|'.join(value))

    return synonym(name, descriptor=property(getter, setter))


def text_date_synonym(name):
    """Converts Y-M-D date strings into datetime objects"""

    def getter(self):
        return getattr(self, name)

    def setter(self, value):
        if isinstance(value, basestring):
            setattr(self, name, datetime.strptime(value, '%Y-%m-%d'))
        else:
            setattr(self, name, value)

    return synonym(name, descriptor=property(getter, setter))


def safe_pickle_synonym(name):
    """Used to store Entry instances into a PickleType column in the database.

    In order to ensure everything can be loaded after code changes, makes sure no custom python classes are pickled.
    """

    def only_builtins(item):
        """Casts all subclasses of builtin types to their builtin python type. Works recursively on iterables.

        Raises ValueError if passed an object that doesn't subclass a builtin type.
        """

        supported_types = [str, unicode, int, float, bool, datetime]
        # dict, list, tuple and set are also supported, but handled separately

        if type(item) in supported_types:
            return item
        elif isinstance(item, dict):
            result = {}
            for key, value in item.iteritems():
                try:
                    result[key] = only_builtins(value)
                except TypeError:
                    continue
            return result
        elif isinstance(item, (list, tuple, set)):
            result = []
            for value in item:
                try:
                    result.append(only_builtins(value))
                except ValueError:
                    continue
            if isinstance(item, list):
                return result
            elif isinstance(item, tuple):
                return tuple(result)
            else:
                return set(result)
        else:
            for s_type in supported_types:
                if isinstance(item, s_type):
                    return s_type(item)

        # If item isn't a subclass of a builtin python type, raise ValueError.
        raise TypeError('%r is not a subclass of a builtin python type.' % type(item))

    def getter(self):
        return getattr(self, name)

    def setter(self, entry):
        setattr(self, name, only_builtins(entry))

    return synonym(name, descriptor=property(getter, setter))


def quality_synonym(name):
    """Get and set property as Quality objects, store as text in the database."""

    def getter(self):
        return qualities.get(getattr(self, name))

    def setter(self, value):
        if isinstance(value, basestring):
            setattr(self, name, value)
        else:
            setattr(self, name, value.name)

    return synonym(name, descriptor=property(getter, setter))


class property_(object):
    """A property for an sqlalchemy model that can also return the proper sql to be used in queries."""

    def __init__(self, fget, fset=None, fdel=None, expr=None):
        self.fget = fget
        self.fset = fset
        self.fdel = fdel
        self.expr = expr or fget
        update_wrapper(self, fget)

    def __get__(self, instance, owner):
        if instance is None:
            return self.expr(owner)
        else:
            return self.fget(instance)

    def __set__(self, instance, value):
        self.fset(instance, value)

    def __delete__(self, instance):
        self.fdel(instance)

    def setter(self, fset):
        self.fset = fset
        return self

    def deleter(self, fdel):
        self.fdel = fdel
        return self

    def expression(self, expr):
        self.expr = expr
        return self


def year_property(date_attr):

    def getter(self):
        date = getattr(self, date_attr)
        return date and date.year

    def expr(cls):
        return extract('year', getattr(cls, date_attr))

    return property_(getter, expr=expr)


def lower_property(str_attr):
    """A property based on a text column that returns or queries on the lower case version of the string."""

    def getter(self):
        string = getattr(self, str_attr)
        return string and string.lower()

    def expr(cls):
        return func.lower(getattr(cls, str_attr))

    return property_(getter, expr=expr)


def quality_comp_property(str_attr):
    """Returns the value for the quality stored, which can be filtered against or used to sort queries."""

    def getter(self):
        return qualities.get(getattr(self, str_attr)).value

    def expr(cls):
        whens = {}
        for qual in qualities.all():
            whens[qual.name] = qual.value
        return case(value=getattr(cls, str_attr), whens=whens)

    return property_(getter, expr=expr)
