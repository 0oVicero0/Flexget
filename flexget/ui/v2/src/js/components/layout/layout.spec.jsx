import React from 'react';
import renderer from 'react-test-renderer';
import Layout from 'components/layout';
import { themed, router, provider } from 'utils/tests';
import fetchMock from 'fetch-mock';

function renderLayout() {
  return provider(router(themed(
    <Layout>
      <div />
    </Layout>
  )), { version: {}, status: { loading: {} } });
}
describe('components/layout', () => {
  beforeEach(() => {
    fetchMock
        .get('/api/server/version', {});
  });

  it('renders correctly', () => {
    const tree = renderer.create(renderLayout()).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

