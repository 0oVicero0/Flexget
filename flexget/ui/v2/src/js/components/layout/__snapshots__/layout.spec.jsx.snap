// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout renders correctly 1`] = `
.glamor-23 {
  color: #b0bec5;
}

.glamor-91 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  min-height: 100vh;
}

.glamor-21 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  min-height: 5rem;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  z-index: 2;
  position: fixed;
  width: 100%;
}

.glamor-3 {
  height: 5rem;
}

.glamor-1 {
  background: #263238 url(test-file-stub) no-repeat center;
  background-size: 17.5rem;
  height: 100%;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-size 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,background-size 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.glamor-19 {
  height: 5rem;
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-17 {
  position: static;
}

.glamor-15 {
  background-color: #ef6c00;
  min-height: 5rem;
}

.glamor-5 {
  color: rgba(255, 255, 255, 1);
}

.glamor-7 {
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.glamor-89 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-top: 10rem;
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100vh;
}

.glamor-85 {
  overflow-y: auto;
}

.glamor-83 {
  position: relative;
  background-color: #263238;
  height: calc(100vh - 10rem);
  width: 0;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-right: none !important;
}

.glamor-82 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-71 {
  width: inherit;
}

.glamor-27 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-27:hover {
  border-left: 3px solid #ff9800;
}

.glamor-80 {
  color: #f44336;
  display: none;
}

.glamor-73 {
  margin: 0px;
}

.glamor-87 {
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow-y: auto;
  padding: 1rem;
  opacity: 1;
  -webkit-transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,margin-left 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,margin-left 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

@media (min-width:600px) {
  .glamor-21 {
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:600px) {
  .glamor-1 {
    width: 6rem;
    background-size: 25rem;
  }
}

@media (min-width:600px) {
  .glamor-89 {
    padding-top: 5rem;
  }
}

@media (min-width:600px) {
  .glamor-83 {
    height: calc(100vh - 5rem);
    width: 6rem;
  }
}

@media (max-width:599.99px) {
  .glamor-87 {
    margin-left: 0px;
    opacity: 1;
    display: block;
  }
}

<div
  className="glamor-90 glamor-91"
>
  <header
    className="glamor-20 glamor-21"
  >
    <div
      className="glamor-2 glamor-3"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="glamor-0 glamor-1"
          open={false}
        />
      </a>
    </div>
    <nav
      className="glamor-18 glamor-19"
    >
      <header
        className="MuiPaper-root-8 MuiPaper-shadow4-14 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorPrimary-6 mui-fixed glamor-16 glamor-17"
      >
        <div
          className="MuiToolbar-root-35 MuiToolbar-gutters-36 glamor-14 glamor-15"
        >
          <button
            className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-4 glamor-5"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-43"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-bars"
              />
            </span>
            <span
              className="MuiTouchRipple-root-55"
            />
          </button>
          <div
            className="glamor-6 glamor-7"
          />
          <a
            href="/config"
            onClick={[Function]}
          >
            <button
              aria-label="Config"
              className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-4 glamor-5"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-43"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-pencil"
                />
              </span>
              <span
                className="MuiTouchRipple-root-55"
              />
            </button>
          </a>
          <a
            href="/log"
            onClick={[Function]}
          >
            <button
              aria-label="Log"
              className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-4 glamor-5"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-43"
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-book"
                />
              </span>
              <span
                className="MuiTouchRipple-root-55"
              />
            </button>
          </a>
          <button
            aria-label="Manage"
            className="MuiButtonBase-root-46 MuiIconButton-root-37 glamor-4 glamor-5"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-43"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-48 MuiIconButton-icon-44 fa fa-cog"
              />
            </span>
            <span
              className="MuiTouchRipple-root-55"
            />
          </button>
        </div>
      </header>
    </nav>
  </header>
  <main
    className="glamor-88 glamor-89"
  >
    <aside
      className="glamor-84 glamor-85"
    >
      <div
        className="MuiDrawer-docked-77"
      >
        <div
          className="MuiPaper-root-8 MuiPaper-shadow0-10 MuiDrawer-paper-72 glamor-83 MuiDrawer-anchorLeft-73"
        >
          <div
            className="glamor-81 glamor-82"
          >
            <ul
              className="MuiList-root-79 MuiList-padding-80 glamor-70 glamor-71"
            >
              <a
                href="/config"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-pencil glamor-23"
                  />
                  <div
                    className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                  >
                    Config
                  </div>
                </li>
              </a>
              <a
                href="/history"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-history glamor-23"
                  />
                  <div
                    className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                  >
                    History
                  </div>
                </li>
              </a>
              <a
                href="/log"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-book glamor-23"
                  />
                  <div
                    className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                  >
                    Log
                  </div>
                </li>
              </a>
              <a
                href="/movies"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-film glamor-23"
                  />
                  <div
                    className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                  >
                    Movies
                  </div>
                </li>
              </a>
              <a
                href="/pending"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-check glamor-23"
                  />
                  <div
                    className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                  >
                    Pending
                  </div>
                </li>
              </a>
              <a
                href="/seen"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-eye glamor-23"
                  />
                  <div
                    className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                  >
                    Seen
                  </div>
                </li>
              </a>
              <a
                href="/series"
                onClick={[Function]}
              >
                <li
                  className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                  disabled={false}
                  onClick={[Function]}
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-tv glamor-23"
                  />
                  <div
                    className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                  >
                    Series
                  </div>
                </li>
              </a>
              <li
                className="MuiListItem-root-83 MuiListItem-gutters-90 MuiListItem-default-86 glamor-26 glamor-27"
                disabled={false}
                onClick={[Function]}
              >
                <span
                  aria-hidden="true"
                  className="material-icons MuiIcon-root-48 glamor-22 MuiListItemIcon-root-92 fa fa-tasks glamor-23"
                />
                <div
                  className="MuiListItemText-root-93 MuiListItemText-inset-94 glamor-24 glamor-23"
                >
                  Tasks
                </div>
              </li>
            </ul>
            <div
              className="glamor-78 glamor-79 glamor-80"
            >
              <p
                className="glamor-72 glamor-73"
              >
                Version Info
              </p>
              <p
                className="glamor-72 glamor-73"
              >
                Flexget: 
                 
                 
              </p>
              <p
                className="glamor-72 glamor-73"
              >
                API: 
              </p>
            </div>
          </div>
        </div>
      </div>
    </aside>
    <section
      className="glamor-86 glamor-87"
      open={false}
    >
      <div />
    </section>
  </main>
</div>
`;
